File: TS_JT-Reader/TS_JT-Reader_Vis-Studio/TS_JT-Reader/MetaDataNode.ts
module C_sharp_JT_Reader {
    export class MetaDataNode extends C_sharp_JT_Reader.GroupJTNode {
        protected encoding: System.Text.Encoding = System.Text.Encoding.Unicode;
        protected elementLength: Int32;
        compare: string;
        _propertyProxyNode: PropertyProxyMetaData;
        m_textBox: List<string> = new List<string>();
        constructor() {

        }
        constructor(fileVersion: number, richTextBox: List<string>) {
            _fileVersion = fileVersion;
            this.m_textBox = richTextBox;
        }
        public populateData(uncompressed: number[], filePosCount: number): number {
            this.m_textBox.Add("\n\n---------------------------------- Meta Data ----------------------------------");
            _uncompressed = uncompressed;
            _filePosCount = filePosCount;
            _filePosCount = TraverseGroupNode(_richTextBox, _filePosCount);
            Buffer.BlockCopy(_uncompressed, _filePosCount, fileBytes, 0, 4);
            _versionNumber = DataTypes.getInt16(fileBytes);
            this.m_textBox.Add("\nVersion number = " + _versionNumber.ToString());
            _filePosCount += 2;
            return this._filePosCount;
        }
    }
}

File: TS_JT-Reader/C#_JT-Reader_Vis-Studio/C sharp JT-Reader/Nodes/MetaDataNode.cs
/////////////////////////////////////////////////////////////////////
//
// This class contains all the Meta Data used by the JT File
//
// Refer to 7.2.6 Meta Data Segment page 162 - on (Ver 9.5 rev D)
//
/////////////////////////////////////////////////////////////////////

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using System.Windows.Forms;

namespace C_sharp_JT_Reader
{
    public class MetaDataNode : C_sharp_JT_Reader.GroupJTNode
    {
        protected System.Text.Encoding encoding = System.Text.Encoding.Unicode;
        protected Int32 elementLength;
        string compare;

        PropertyProxyMetaData _propertyProxyNode;
        List<string> m_textBox = new List<string>();

        public MetaDataNode()
        {
        }

        public MetaDataNode(float fileVersion, List<string> richTextBox)
        {
            _fileVersion = fileVersion;
            m_textBox = richTextBox;
        }

        public int populateData(byte[] uncompressed, int filePosCount)
        {
            this.m_textBox.Add("\n\n---------------------------------- Meta Data ----------------------------------");

            _uncompressed = uncompressed;
            _filePosCount = filePosCount;


            // Traverse the Group Node Data
            _filePosCount = TraverseGroupNode(_richTextBox, _filePosCount);

            //_filePosCount -= 8;// 6;//9


            Buffer.BlockCopy(_uncompressed, _filePosCount, fileBytes, 0, 4);
            _versionNumber = DataTypes.getInt16(fileBytes);
            this.m_textBox.Add("\nVersion number = " + _versionNumber.ToString());
            _filePosCount += sizeof(Int16);

            return this._filePosCount;
        }
    }
}
